nucliadb/nucliadb/ingest/orm/resource.py

```diff
@@ -327,7 +327,7 @@ async def set_relations(self, payload: list[PBRelation]):
         self.relations = relations
 
     @processor_observer.wrap({"type": "generate_index_message"})
-    async def generate_index_message(self) -> ResourceBrain:
+    async def generate_index_message(self, reindex: bool = False) -> ResourceBrain:
```

Comment:
I wonder if reindex should always be true.
is there any occurrence of it being false?

Reply:
Yes, it's set to false during rollover and rebalance, where new resources are created without the need to delete old data.